src/jit/lclvars.cpp

```diff
@@ -3363,15 +3363,20 @@ void Compiler::lvaAllocOutgoingArgSpace()
         lvaTable[lvaOutgoingArgSpaceVar].lvRefCnt     = 1;
         lvaTable[lvaOutgoingArgSpaceVar].lvRefCntWtd  = BB_UNITY_WEIGHT;
 
-#if defined(PROFILING_SUPPORTED) && defined(_TARGET_AMD64_) && !defined(UNIX_AMD64_ABI) // No 4 slots for outgoing params on System V.
+        if (compUsesThrowHelper && (lvaOutgoingArgSpaceSize == 0))
+        {
+            // For Windows Amd64 we guarantee that if there is a call we allocate the
+            // ABI mandated 4 slots outgoing arg space.
+            lvaOutgoingArgSpaceSize = MIN_ARG_AREA_FOR_CALL;
+        }
+
         // If we are generating profiling Enter/Leave/TailCall hooks, make sure
         // that outgoing arg space size is minimum 4 slots.  This will ensure
         // that even methods without any calls will have 4-slot outgoing arg area.
         if (compIsProfilerHookNeeded() && (lvaOutgoingArgSpaceSize == 0))
         {
-            lvaOutgoingArgSpaceSize = 4 * REGSIZE_BYTES;            
+            lvaOutgoingArgSpaceSize = MIN_ARG_AREA_FOR_CALL;
```

Comment:
Note that this logic is already under #if FEATURE_FIXED_OUT_ARGS, which is true for Amd64 unix/windiws, Arm64 and arm32.  But only Amd64 windows requires this logic.  Though this doesn't pose a correctness issue, there is no reason to enable to this logic for other arch where it is not required.  I would suggest to include this logic under #if MIN_ARG_AREA_FOR_CALL != 0

e.g. 
```
#if MIN_ARG_AREA_FOR_CALL != 0
if (lvaOutgoingArgSpaceSize  == 0)
{
     // comment lines go here
     if (compUsesThrowHelper  || compIsProfilerHookNeeded())
         lvaOutgoingArgSpaceSize = MIN_ARG_AREA_FOR_CALL;
}
#endif
```